modules/exploits/multi/http/struts_code_exec_classloader.rb

```diff
@@ -265,11 +271,8 @@ def class_loader_exploit
       fail_with(Failure::Unknown, "#{peer} - The log file hasn't been flushed")
     end
 
-    if target['Platform'] == 'win' && target['Arch'] == ARCH_X86
-      register_files_for_cleanup("../webapps/ROOT/#{@jsp_file}")
-    else
-      register_files_for_cleanup(@jsp_file)
-    end
+    # This path depends on CWD. May require manual cleanup
+    register_files_for_cleanup("webapps/ROOT/#{@jsp_file}")
```

Comment:
This is what is unreliable, depends on CWD.
As said would be nice to fix https://github.com/rapid7/metasploit-framework/issues/4667. In that case at least the user would know that file hasn't been deleted